completable.go

```diff
@@ -0,0 +1,13 @@
+package sqlds
+
+import (
+	"context"
+	"database/sql"
+)
+
+// Completable will be used to autocomplete Tables Schemas and Columns for SQL languages
+type Completable interface {
+	Tables(ctx context.Context, db *sql.DB) ([]string, error)
+	Schemas(ctx context.Context, db *sql.DB) ([]string, error)
+	Columns(ctx context.Context, db *sql.DB, table string) ([]string, error)
```

Comment:
I'm not sure that having a `db *sql.DB` argument is necessary on these, it might make writing a `Test` version of this interface kind of difficult. Most things that satisfy this interface will have a `db` field in the struct